src/main/java/dk/kb/image/api/v1/impl/AccessApiServiceImpl.java

```diff
@@ -395,6 +408,77 @@ public StreamingOutput iIPImageRequest(
         }
     }
      
+    /* Is request classified as a thumbnail or fullsize for IIIF requests.
+     * 
+     * This implementation is very conservative and will determine thumbnail also if most non size-parameters are defined.
+     * It is better to be conservative and later loosen up than giving too much control over thumbnail extraction.
+     * 
+     * Will be full size if any other parameters than FIF and CVT is defined. Also WID and HEI must be below a defined limit in the configuration or it will also be fullsize.   
+     */
+    private boolean isThumbnailIIIF(String identifier, String region, String size, String rotation, String quality, String format) {
+         if (!"full".equals(region)  ||  !rotation.equals("0") || !quality.equals("default")) {        	         	
+        	 log.debug("Fullsize for IIIF request since a custom parameter was defined");             
+             return false;                
+         }
+         else if (size == null) {
+             log.debug("Fullsize for IIIF request since size parameter was not defined");
+             return false;
+         }                           
+         
+            //Only allow size as "w,h" parameter. Etc. "100,100". Value is default 'max' when requesting the full image.
+         String[] tokens = size.substring(1).split(","); //First character is !
+         if (tokens.length != 2) {
+             log.debug("Fullsize for IIIF request since size parameter not width,hight");                          
+             return false;
+         }
+         int width;
+         int height;         
+         try {
+             width=Integer.parseInt(tokens[0].trim());
+             height=Integer.parseInt(tokens[1].trim());
+         }
+         catch(Exception e) {
+             log.debug("Fullsize for IIIF request since size parameter could not be parsed as (width,height) integers:"+size);
+             return false;
+         }
+                  
+        if ( width > ServiceConfig.getConfig().getInteger("thumbnail.maxWidth") || height > ServiceConfig.getConfig().getInteger("thumbnail.maxHeight")){         
+             log.debug("Fullsize for IIIF request since size parameter was over thumbnail size");             
```

Comment:
It helps a lot to have some sort of context when a flag is raised. Please the `identifier` to all log messages in `idThumbnailIIIF` and `isThumbnailIIP`. In the case of checks where `width` or `height` exceeds the maxima, logging the provided size also helps when looking through the logs.

Reply:
Added

Reply:
Looks good now, except for a slight error in line 101
```Java
             log.debug("Fullsize for IIIF request since size parameter could not be parsed as (width,height) integers value={} , identifier={}"+size,identifier);
```
(`+` instead of `,`)

Reply:
Fixed.. could IDEA help see it? (brackets not matching arguments)

Reply:
IDEA complained, yes.